rpc/src/services/protocol/mod.rs

```diff
@@ -427,63 +428,72 @@ pub(crate) fn proto_get_contract_manager_key(
                 pkh,
                 context)
         }
-        _ => panic!("Missing endorsing rights implemetation for protocol: {}, protocol is not yet supported!", hash)
+        _ => panic!("Missing manager key implemetation for protocol: {}, protocol is not yet supported!", hash)
     }
 }
 
 pub(crate) fn run_operation(chain_param: &str, block_param: &str, json_request: JsonRpcRequest, env: &RpcServiceEnvironment) -> Result<serde_json::value::Value, failure::Error> {
     let persistent_storage = env.persistent_storage();
     let state = env.state();
 
-    // get header
-    let block_storage = BlockStorage::new(persistent_storage);
-    let block_hash = get_block_hash_by_block_id(block_param, persistent_storage, state)?;
-    let block_header = block_storage.get(&block_hash)?;
-    let block_header = match block_header {
-        Some(header) => header.header.as_ref().clone(),
-        None => bail!("No block header found for hash: {}", block_param)
-    };
-    let state = state.read().unwrap();
-    let chain_id = state.chain_id().clone();
+    let request = create_protocol_json_rpc_request(chain_param, block_param, json_request, FfiRpcService::HelpersRunOperation, persistent_storage, state)?;
 
-    // create request to ffi
-    let request = ProtocolJsonRpcRequest {
-        chain_arg: chain_param.to_string(),
-        block_header,
-        ffi_service: FfiRpcService::HelpersRunOperation,
-        request: json_request,
-        chain_id,
-    };
+    // TODO: retry?
+    let response = env.tezos_readonly_api().pool.get()?.api.call_protocol_json_rpc(request)?;
+
+    Ok(serde_json::from_str(&response.body)?)
+}
+
+pub(crate) fn current_level(chain_param: &str, block_param: &str, json_request: JsonRpcRequest, env: &RpcServiceEnvironment) -> Result<serde_json::value::Value, failure::Error> {
+    let persistent_storage = env.persistent_storage();
+    let state = env.state();
+
+    let request = create_protocol_json_rpc_request(chain_param, block_param, json_request, FfiRpcService::HelpersCurrentLevel, persistent_storage, state)?;
```

Comment:
@adonagy 
maybe, you could also send to `create_protocol_json_rpc_request` just reference to env, instead of persistent_storage and state